docs/upgrade-smart-agent.md

```diff
@@ -0,0 +1,179 @@
+
+1. Upgrade to the latest version:
+
+Chef:
+ * 1.0
+
+Ansible:
+ * https://github.com/signalfx/signalfx-agent/tree/v5.0.0/deployments/ansible
+
+Puppet:
+ * 1.0
+
+Salt:
+ * https://github.com/signalfx/signalfx-agent/tree/v5.0.0/deployments/salt
+
+2. Verify that the old keys were removed. Based on how you installed the agent, there are two method:
+  * Method 1: APT
+    1. Run the following command: 
+    ```sh
+    $ apt-key list
+    ```
+    2. Verify that the key ending in 5AE495F6 is not present. You should see 58C3 3310 B7A3 54C1 279D  B669 5EFA 01ED B3CD 4420.
+
+  * Method 2: RPM
+    1. Run the following command: 
+      ```sh
+      $ rpm -q gpg-pubkey --qf '%{NAME}-%{VERSION}-%{RELEASE}\t%{SUMMARY}\n'
+      ```
+    2. Verify that the gpg-pubkey-098acf3b-55a5351a key is not present. You should only see gpg-pubkey-b3cd4420-5b5b79b1.
+
+## Additional information
+
+### Continue with Smart Agent 4.X
+
+If you want to continue with Smart Agent 4.X, you must pin the agent version. Afterwards, you must run a check to verify that the old key was removed.
+
+Note: Current users can continue to run Smart Agent 4.X; however, after April 21, 2020, any new installs or upgrades will require you to change to the new Splunk-signed repositories.
+
+#### Method 1: APT
+
+1. Delete the existing SignalFx apt key. (You must run this command as root.)
+```sh
+apt-key del 5AE495F6
+```
+
+2. Import the new key. (You must run this command as root.)
+```sh
+curl https://dl.signalfx.com/splunk-B3CD4420.gpg | apt-key add -
+```
+
+3. Replace /etc/apt/sources.list.d/signalfx-agent.list with the contents in the following file:
+```sh
+deb  https://dl.signalfx.com/debs/signalfx-agent/release /
+```
+
+4. Update the package metadata. (You must run this command as root.)
+```sh
+apt-get update
+```
+
+5. Pin the agent version.
+  * To learn more, see https://github.com/signalfx/signalfx-agent/tree/master/deployments.
+
+6. Run the following command:
+```sh
+$ apt-key list
+```
+
+7. Verify that the key ending in 5AE495F6 is not present.
+  * You should see 58C3 3310 B7A3 54C1 279D  B669 5EFA 01ED B3CD 4420.
+
+#### Method 2: RPM
+
+1. Delete the existing SignalFx RPM key. (You must run this command as root.)
+```sh
+rpm -e gpg-pubkey-098acf3b-55a5351a
+```
+
+2. Modify /etc/yum.repos.d/signalfx-agent.repo. Specifically: 
+  * Change baseurl to https://dl.signalfx.com/rpms/signalfx-agent/release
+  * Change gpgkey to https://dl.signalfx.com/splunk-B3CD4420.pub
+
+3. Pin the agent version.
+  * To learn more, see https://github.com/signalfx/signalfx-agent/tree/master/deployments.
+
+4. Run the following command:
+```sh
+$ rpm -q gpg-pubkey --qf '%{NAME}-%{VERSION}-%{RELEASE}\t%{SUMMARY}\n'
+```
+
+5. Verify that the gpg-pubkey-098acf3b-55a5351a key is not present.
+  * You should only see gpg-pubkey-b3cd4420-5b5b79b1.
+
```

Comment:
We need method 3 above but for 4.x along with the note about pinning the agent version.